k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -3037,40 +3040,75 @@ public boolean messagesPendingSend(final Account account) {
      * Attempt to send any messages that are sitting in the Outbox.
      * @param account
      */
+    private LocalFolder localFolder = null;
+    private Exception lastFailure = null;
+    private boolean wasPermanentFailure = false;
+    private List<LocalMessage> localMessages;
+    private LocalStore localStore;
+    private Transport transport;
+    private FetchProfile fp;
```

Comment:
I don't think we should be keeping reference to these. If you need them in sub-methods, pass them in as arguments. If we keep them, we're implying they're stateful.